src/map/map.cpp

```diff
@@ -234,6 +234,10 @@ void map_destroyblock( block_list* bl ){
 	}
 
 	switch( bl->type ){
+		case BL_NUL:
+			// Dummy type, has no destructor
```

Comment:
After seeing the crash reports, I think the whole cause of this is that the destructor sets the BL type to BL_NUL, so if you call the destructor a second time you get the warning.

I'm wondering if we should really suppress this warning then as it could give us necessary information that a destructor was called somewhere were it's not supposed to.

Maybe we could have a message in debug mode here that tells us to check for bl destructor calls?